packages/vxrn/src/utils/swapPrebuiltReactModules.ts

```diff
@@ -231,7 +231,7 @@ export async function swapPrebuiltReactModules(
     async load(id) {
       if (id.startsWith('virtual:rn-internals')) {
         const idOut = id.replace('virtual:rn-internals:', '')
-        let out = `const ___val = __RN_INTERNAL_MODULE_REQUIRES_MAP__["${idOut}"]()
+        let out = (id.includes('codegenNativeComponent') ? `const ___val = __RN_INTERNAL_MODULE_REQUIRES_MAP__["${idOut}"]()` : `const ___val = __cachedModules["${idOut}"]`) /* IDK why we can't use __RN_INTERNAL_MODULE_REQUIRES_MAP__ for all modules, if we do so the app will hang with a white screen (but won't crash) */ + `
```

Comment:
This is really confusing, trying to figure out what's going on.

Reply:
`examples/one-basic` seems to not have this problem but `tests/test`.

Reply:
The actual error is:

```
◆ ./ReactFabric2.js Error running module (parent: ./ReactFabric2.js): _virtual/ReactFabric2.js
Property 'nativeFabricUIManager' doesn't exist
ReferenceError: Property 'nativeFabricUIManager' doesn't exist
```

Reply:
After try-error, it seems that

```js
__cachedModules["react-native/Libraries/Renderer/shims/ReactFabric"]
```

can't be replaced by

```js
__RN_INTERNAL_MODULE_REQUIRES_MAP__["react-native/Libraries/Renderer/shims/ReactFabric"]()
```

Reply:
Which is weird since:

```js
        var __require = function __require2() {
            var cachedMod = globalThis["__cachedModules"][modulePath];
            if (cachedMod) return cachedMod;

            if (mod) return mod;

            var moduleFn = cb[path];
            mod = {
                exports: {}
            };
            moduleFn(mod.exports, mod);
            mod = mod.exports;
            globalThis["__cachedModules"][modulePath] = mod;
            return mod;
        };
        if (globalThis["__RN_INTERNAL_MODULE_REQUIRES_MAP__"]) {
            globalThis["__RN_INTERNAL_MODULE_REQUIRES_MAP__"][modulePath] = __require;
        }
```

As the first few lines it'll just return `__cachedModules["react-native/Libraries/Renderer/shims/ReactFabric"]`.

Reply:
`__cachedModules["react-native/Libraries/Renderer/shims/ReactFabric"]` will be undefined.

So it **should** be undefined instead of loaded I think.

Reply:
```js
                if (useFabric) {
                    require_ReactFabric().render(element, rootTag, null, useConcurrentRoot);
                } else {
                    require_ReactNative().render(element, rootTag);
                }
```

Yes, so I think `ReactFabric` shouldn't be loaded if not already loaded (since such a case, it means that it shouldn't be loaded at all).

Reply:
Ok the problem is that this code, in `tamagui/code/ui/portal/dist/esm/Portal.native.js`:

```js
isFabric ? createPortal = require("react-native/Libraries/Renderer/shims/ReactFabric").createPortal : createPortal = require("react-native/Libraries/Renderer/shims/ReactNative").createPortal;
```

Is being into as this in the react natvie bundle:

```js
const ReactFabric_commonjsProxy = require('../../../../../../../_virtual/ReactFabric.js');
const ReactNative_commonjsProxy = require('../../../../../../../_virtual/ReactNative.js');
// ...
isFabric ? createPortal = ReactFabric_commonjsProxy.default.createPortal : createPortal = ReactNative_commonjsProxy.default.createPortal;
```

With the bundled version, it'll try to require `ReactFabric` no matter `isFabric` is `true` or not.

But if `isFabric` is not `true`, seems that `ReactFabric` shouldn't be required, as doing so will break the app.

Previously, that `require('../../../../../../../_virtual/ReactFabric.js')` line there won't cause a problem since in our virtual module, we're only trying to return the module from a cache:

```js
__cachedModules["react-native/Libraries/Renderer/shims/ReactFabric"]
```

Since the `ReactFabric` module won't be used by RN internally, `__cachedModules[".../ReactFabric"]` will just be `undefined` and `require('.../ReactFabric.js')` won't try to load it, so we're safe here.
`

But now, since [we turn RN exports back into being lazy loaded, we're also replacing `__cachedModules["..."]` into `__RN_INTERNAL_MODULE_REQUIRES_MAP__["..."]()`, where the latter one will try to load the module if it isn't already loaded](https://github.com/onejs/one/pull/283/commits/688c3ec86392f9cbfc08c09178144fb75b330e1b), now `require('.../ReactFabric.js')` will try to load ReactFabric and causes the app to hang in some way.